source/posix/atomic.c

```diff
@@ -0,0 +1,34 @@
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#if defined(__GNUC__) || defined(__clang__)
```

Comment:
Do you think it's a good idea to add more folders for different compilers? These compilers are typically the ones on posix systems, right? Adding more folders for different compilers sounds like it would get real annoying real fast. Would be good to talk about this more in-depth.

Reply:
:+1: to .inl files. These are all compiler intrinsics, and will likely only result in one generated instruction. Adding the overhead of a function call for 1 instruction seems crazy.